### PR TITLE
MOD: support stop reload reopen in cmdline

### PR DESCRIPTION
**Pull request by karelrooted**
usage: nutcracker -k "shutdown/reload/stop/reopen"

### CODE REVIEW

File: src/nc.c [hunk 1]
```diff
@@ -261,6 +266,40 @@ nc_create_pidfile(struct instance *nci)
...
+    if (fd < 0) {
+        log_error("opening pid file '%s' failed: %s", nci->pid_filename,
+                  strerror(errno));
+        return 0;
+    }
+
+    n = nc_read(fd, &pid, NC_UINTMAX_MAXLEN);
+    if (n <= 0) {
+        log_error("Read pid file '%s' failed: %s", nci->pid_filename,
+                  strerror(errno));
```
Comment (git-hulk) [inline src/nc.c]: FIX: close(fd)

File: src/nc.c [hunk 2]
```diff
@@ -261,6 +266,40 @@ nc_create_pidfile(struct instance *nci)
...
+        return 0;
+    }
+
+    pid_value = nc_atoi(pid, strlen(pid));
+    if (pid_value < 0) {
+        log_error("Read pid file '%s' failed: %s", nci->pid_filename,
+                  strerror(errno));
```
Comment (git-hulk) [inline src/nc.c]: ditto

File: src/nc.c [hunk 3]
```diff
@@ -261,6 +266,40 @@ nc_create_pidfile(struct instance *nci)
...
+        return 0;
```
Comment (git-hulk) [inline src/nc.c]: 0 => -1 while the pid 0 is init process

File: src/nc.c [hunk 4]
```diff
@@ -261,6 +266,40 @@ nc_create_pidfile(struct instance *nci)
...
+    }
+
+    n = nc_read(fd, &pid, NC_UINTMAX_MAXLEN);
+    if (n <= 0) {
+        log_error("Read pid file '%s' failed: %s", nci->pid_filename,
+                  strerror(errno));
+        return 0;
+    }
+
```
Comment (git-hulk) [inline src/nc.c]: add `pid[n] = '\0';`  or nc_atoi(pid, n)

File: src/nc.c (hunk 1 above)
Comment (karelrooted) [reply src/nc.c]: fixed

File: src/nc.c (hunk 4 above)
Comment (karelrooted) [reply src/nc.c]: fixed

File: src/nc.c [hunk 5]
```diff
@@ -528,6 +586,34 @@ nc_run(struct instance *nci)
...
 }
 
+static int
+nc_handle_user_cmd(struct instance *nci, const char *cmd) {
+    int signal = 0;
+    unsigned long i;
+
+    for(i = 0; i < sizeof(command_signals)/sizeof(struct command_signal); i++) {
+        if (nc_strncmp( cmd, command_signals[i].command, nc_strlen(cmd) ) == 0) {
+            signal = command_signals[i].signal;
```
Comment (git-hulk) [inline src/nc.c:598]: break;

File: src/nc.c [hunk 6]
```diff
@@ -51,7 +52,19 @@ static int show_version;
...
+    const char *command;
+    int signal;
+};
+
+static struct command_signal command_signals[] = {
+    { "reload",    SIGHUP  },
+    { "reopen",    SIGUSR1 },
+    { "stop",      SIGTERM },
+    { "shutdown",  SIGTERM }
+};
```
Comment (git-hulk) [inline src/nc.c:67]: +blank

File: src/nc.c [hunk 7]
```diff
@@ -587,7 +587,20 @@ nc_run(struct instance *nci)
 }
 
 static int
```
Comment (git-hulk) [inline src/nc.c]: static int => static void, and replace return with exit()

File: src/nc.c (hunk 5 above)
Comment (karelrooted) [reply src/nc.c:598]: fixed

File: src/nc.c (hunk 7 above)
Comment (karelrooted) [reply src/nc.c]: fixed

**Review by git-hulk — APPROVED** [review]

**Review by git-hulk — COMMENTED** [review]
Rebase the commits

**Comment by karelrooted** [comment]
the commits has been rebased into one commit,please review the new commit

**Pull request merged**